types/tx.go

```diff
@@ -11,19 +11,51 @@ import (
 	tmproto "github.com/lazyledger/lazyledger-core/proto/tendermint/types"
 )
 
-// Tx is an arbitrary byte array.
+// Tx has a key and value arbitrary byte array.
 // NOTE: Tx has no types at this level, so when wire encoded it's just length-prefixed.
 // Might we want types here ?
-type Tx []byte
+type Tx struct {
+	Key   []byte
+	Value []byte
+}
 
 // Hash computes the TMHASH hash of the wire encoded transaction.
+// Hash(key) + Hash(value)
 func (tx Tx) Hash() []byte {
-	return tmhash.Sum(tx)
+	var hash []byte
```

Comment:
This may be better as a hash of the value. This way we can still support get tx by hash as it currently stands.